### PR TITLE
added shape property to SliverAppBar

### PR DESCRIPTION
**Pull request by mtiziano**
re-open PR https://github.com/flutter/flutter/pull/31662 with @shihaohong 

### Description
added **shape** property in **SilverAppBar** widget, which was already implemented in **AppBar** widget

### Related Issues
#31656

### CODE REVIEW

**Comment by mtiziano** [comment]
CLA updated

File: packages/flutter/test/material/app_bar_test.dart [hunk 1]
```diff
@@ -1514,4 +1514,39 @@ void main() {
...
+              shape: roundedRectangleBorder,
+              actions: <Widget>[Text('A1'), Text('A2')],
+            ),
+          ],
+        ),
+      ),
+    );
+
+    final Finder sliverAppBarFinder = find.byType(SliverAppBar);
+
```
Comment (shihaohong) [inline packages/flutter/test/material/app_bar_test.dart]: Remove newline to keep formatting consistent (there is no space between line 1545 and 1546 below)

File: packages/flutter/test/material/app_bar_test.dart [hunk 2]
```diff
@@ -1514,4 +1514,39 @@ void main() {
...
     expect(getMaterialWidget().shape, roundedRectangleBorder);
   });
+
+  testWidgets('SliverAppBar with shape', (WidgetTester tester) async {
+    const RoundedRectangleBorder roundedRectangleBorder = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(15.0)),
+    );
+    await tester.pumpWidget(
+      MaterialApp(
+        home: CustomScrollView(
```
Comment (shihaohong) [inline packages/flutter/test/material/app_bar_test.dart]: Your `const` placement in this block fails `flutter analyze --flutter-repo`. Please make sure analyzer tests are passing.

File: packages/flutter/test/material/app_bar_test.dart (hunk 1 above)
Comment (shihaohong) [reply packages/flutter/test/material/app_bar_test.dart]: Could you also remove line 1503 as well?

**Review by shihaohong — COMMENTED** [review]
A few comments about formatting. Also, this could be a good time to break down line 1490 into three lines if you can! 

Make sure to pass the analyzer test as well, since that's the main reason the original PR was reverted.

**Comment by shihaohong** [comment]
Closing this PR out, as it has been resolved with https://github.com/flutter/flutter/pull/33073. Thanks for your contribution @mtiziano

**Pull request closed without merging**